sensor/upgrader/preflight/schema_validation.go

```diff
@@ -15,23 +22,27 @@ func (schemaValidationCheck) Name() string {
 }
 
 func (schemaValidationCheck) Check(ctx *upgradectx.UpgradeContext, execPlan *plan.ExecutionPlan, reporter checkReporter) error {
-	jsonEncoder := json.NewSerializer(json.DefaultMetaFactory, nil, nil, false)
-
 	for _, act := range execPlan.Actions() {
 		if act.Object == nil {
 			continue
 		}
 
-		var buf bytes.Buffer
-		if err := jsonEncoder.Encode(act.Object, &buf); err != nil {
-			reporter.Errorf("Failed to serialize object %v to JSON: %v", act.ObjectRef, err)
-			continue
+		if err := validateObject(act.Object, ctx.Validator()); err != nil {
+			reporter.Errorf("Object %s: %s", act.ObjectRef, err)
 		}
+	}
 
-		if err := ctx.Validator().ValidateBytes(buf.Bytes()); err != nil {
-			reporter.Errorf("Schema validation for object %v failed: %v", act.ObjectRef, err)
-		}
+	return nil
+}
+
+func validateObject(obj k8sutil.Object, validator validation.Schema) error {
+	var buf bytes.Buffer
+	if err := defaultJSONEncoder.Encode(obj, &buf); err != nil {
+		return fmt.Errorf("failed to serialize to JSON: %w", err)
```

Comment:
Yeah, not sure why I went with `fmt.Errorf` here. You're completely right about the consistency aspect. However, since the error is wrapped (and `errors.Errorf` doesn't support this), the correct function to use is `errors.Wrap` (inconsequential because of the use in `Reportf`, but still)